pkg/build/builder/docker.go

```diff
@@ -117,18 +119,20 @@ func (d *DockerBuilder) Build() error {
 
 			timing.RecordNewStep(ctx, buildapiv1.StagePullImages, buildapiv1.StepPullBaseImage, startTime, metav1.Now())
 
-			if err != nil {
+			// if we are unable to pull the image, and the name
+			// of the image that we tried to pull is the same as
+			// the name of a stage in the dockerfile, ignore the error
```

Comment:
why wouldn't we check if the image name is the name of a previous stage before attempting to pull it?

Reply:
Well, I had it like that before and you said it could cause x image to not get pulled, so i changed it ... o_O